# Calculate slope between initiation and start of the bloom peak ($m$)

# First, load libraries

In [1]:
import numpy as np
import pandas as pd

# Slope

tI: date of the bloom initiation (day of year, 1-based).

cI: vertically integrated Chl $a$ just before the bloom initiation (0-100m; $mg\ Chl\ m^{-3}$).

tP: date of the start of the bloom peak period (day of year, 1-based).

cP: vertically integrated Chl a at tP (0-100m; $mg\ Chl\ m^{-3}$).

m: slope of of IP.

$m=\frac{log(c_P)-log(c_I)}{t_P-t_I}$

# Slopes in the model world

In [2]:
d={'simulation':['EXP-0','EXP-1.1','EXP-1.2','EXP-1.3.0_25','EXP-1.3.0.50','EXP-1.3.2_00','EXP-1.3.4_00'],
   'tI'        :[ 163,    201,      165,      163,           163,           163,           163],
   'cI'        :[ 2.467,  1.878,    0.0002427,2.518,         2.506,         2.504,         2.504],
   'tP'        :[ 198,    225,      228,      191,           194,           198,           198],
   'cP'        :[ 40.311, 73.113,   43.370,   25.567,        34.789,        41.044,        41.223]
  }
simulations_df=pd.DataFrame(data=d)

In [3]:
d

{'simulation': ['EXP-0',
  'EXP-1.1',
  'EXP-1.2',
  'EXP-1.3.0_25',
  'EXP-1.3.0.50',
  'EXP-1.3.2_00',
  'EXP-1.3.4_00'],
 'tI': [163, 201, 165, 163, 163, 163, 163],
 'cI': [2.467, 1.878, 0.0002427, 2.518, 2.506, 2.504, 2.504],
 'tP': [198, 225, 228, 191, 194, 198, 198],
 'cP': [40.311, 73.113, 43.37, 25.567, 34.789, 41.044, 41.223]}

In [4]:
simulations_df

simulation   tI        cI   tP      cP
0         EXP-0  163  2.467000  198  40.311
1       EXP-1.1  201  1.878000  225  73.113
2       EXP-1.2  165  0.000243  228  43.370
3  EXP-1.3.0_25  163  2.518000  191  25.567
4  EXP-1.3.0.50  163  2.506000  194  34.789
5  EXP-1.3.2_00  163  2.504000  198  41.044
6  EXP-1.3.4_00  163  2.504000  198  41.223

In [5]:
simulations_df['m']=( np.log10(simulations_df['cP'])-np.log10(simulations_df['cI']) ) \
/(simulations_df['tP']-simulations_df['tI'])

Table 1. Features of the simulated blooms. tI is the date of the bloom initiation (day of year, 1-based). cI is the vertically integrated Chl $a$ just before the bloom initiation (0-100m; $mg\ Chl\ m^{-3}$). tP is the date of the start of the bloom peak period (day of year, 1-based). cP is the vertically integrated Chl a at tP (0-100m; $mg\ Chl\ m^{-3}$). m is the slope of of IP.

In [6]:
simulations_df

simulation   tI        cI   tP      cP         m
0         EXP-0  163  2.467000  198  40.311  0.034664
1       EXP-1.1  201  1.878000  225  73.113  0.066262
2       EXP-1.2  165  0.000243  228  43.370  0.083367
3  EXP-1.3.0_25  163  2.518000  191  25.567  0.035951
4  EXP-1.3.0.50  163  2.506000  194  34.789  0.036854
5  EXP-1.3.2_00  163  2.504000  198  41.044  0.034703
6  EXP-1.3.4_00  163  2.504000  198  41.223  0.034757

In [7]:
simulations_df['m'].describe()

count    7.000000
mean     0.046651
std      0.019879
min      0.034664
25%      0.034730
50%      0.035951
75%      0.051558
max      0.083367
Name: m, dtype: float64

The range of values for the slope in the simulations was 0.035 to 0.083. The mean was 0.047 and the standard deviation was 0.020.

Since these values were calculated from simulations, I would not use them for predictions.


# Slopes in the real world

In [8]:
infileGEic2015='../../../obs/GEic2015_ti_ci_tp_cp.csv'
infileGEic2016='../../../obs/GEic2016_ti_ci_tp_cp.csv'

In [9]:
GEic2015_df=pd.read_csv(infileGEic2015)
GEic2015_df.insert(loc=0,column='obs',value='Qikiqtarjuaq2015')
GEic2016_df=pd.read_csv(infileGEic2016)
GEic2016_df.insert(loc=0,column='obs',value='Qikiqtarjuaq2016')
obs_df=pd.concat([GEic2015_df,GEic2016_df])
obs_df

obs   tI        cI   tP          cP
0  Qikiqtarjuaq2015  158  6.280748  195  314.640619
0  Qikiqtarjuaq2016  148  5.506525  197  204.442579

In [10]:
obs_df['m']=( np.log10(obs_df['cP'])-np.log10(obs_df['cI']) ) \
/(obs_df['tP']-obs_df['tI'])

Table 1. Features of the observed blooms. tI is the date of the bloom initiation (day of year, 1-based). cI is the vertically integrated Chl $a$ just before the bloom initiation (0-100m; $mg\ Chl\ m^{-3}$). tP is the date of the start of the bloom peak period (day of year, 1-based). cP is the vertically integrated Chl a at tP (0-100m; $mg\ Chl\ m^{-3}$). m is the slope of of IP.

In [11]:
obs_df

obs   tI        cI   tP          cP         m
0  Qikiqtarjuaq2015  158  6.280748  195  314.640619  0.045941
0  Qikiqtarjuaq2016  148  5.506525  197  204.442579  0.032035

In [12]:
obs_df['m'].describe()

count    2.000000
mean     0.038988
std      0.009833
min      0.032035
25%      0.035511
50%      0.038988
75%      0.042464
max      0.045941
Name: m, dtype: float64

The range of values for the slope in the observations was 0.032 to 0.046. The mean was 0.039 and the standard deviation was 0.0098.